Modules/_testcapi/long.c

```diff
@@ -141,13 +141,155 @@ pylong_aspid(PyObject *module, PyObject *arg)
 }
 
 
+static PyObject *
+layout_to_dict(const PyLongLayout *layout)
+{
+    PyObject *dict = PyDict_New();
+    if (dict == NULL) {
+        goto error;
+    }
+
+#define SET_DICT(KEY, EXPR) \
+    do { \
+        PyObject *value = (EXPR); \
+        if (value == NULL) { \
+            goto error; \
+        } \
+        int res = PyDict_SetItemString(dict, KEY, value); \
+        Py_DECREF(value); \
+        if (res < 0) { \
+            goto error; \
+        } \
+    } while (0)
+
+    SET_DICT("bits_per_digit", PyLong_FromUnsignedLong(layout->bits_per_digit));
+    SET_DICT("digit_size", PyLong_FromUnsignedLong(layout->digit_size));
+    SET_DICT("digits_order", PyLong_FromLong(layout->digits_order));
+    SET_DICT("digit_endianness", PyLong_FromLong(layout->digit_endianness));
+#undef SET_DICT
+
+    return dict;
+
+error:
+    Py_XDECREF(dict);
+    return NULL;
+}
+
+
+static PyObject *
+pylong_export(PyObject *module, PyObject *obj)
+{
+    PyLongExport export_long;
+    if (PyLong_Export(obj, &export_long) < 0) {
+        return NULL;
+    }
+
+    if (export_long.digits == NULL) {
+        return PyLong_FromInt64(export_long.value);
```

Comment:
You need to test also `negative` and `ndigits` in this case.

Reply:
I added assertions (which are always built).